### PR TITLE
Finished implementing user preferences repository

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/114" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/RepositoryModule.java [hunk 1]
```diff
@@ -42,9 +43,11 @@ ImagesRepository provideImageRepository(Context context,
 
     @Provides
     @Singleton
-    UserPreferencesRepository providePreferencesRepository(
-            PreferencesProvider preferencesProvider) {
-        return new UserSettingsRepository(preferencesProvider);
+    UserPreferencesRepository provideUserPreferencesRepository(Context context) {
+        PreferenceManager.setDefaultValues(context, R.xml.pref_general, false);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/RepositoryModule.java]: EWeweweweweweweweweweweeewwwwwww.
That's ugly :<

Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/RepositoryModule.java]: default values should be handled elsewhere.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java [hunk 2]
```diff
@@ -40,6 +40,9 @@
     @Inject
     LocationFetcher mLocationFetcher;
 
+    @Inject
+    UserPreferencesRepository userPreferencesRepository;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: It seems like this field isn't used.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java [hunk 3]
```diff
@@ -90,6 +87,9 @@
     @Inject
     SyncDataConnectivityStatusInteractorFactory mDataAlertsFactory;
 
+    @Inject
+    UserPreferencesRepository userPreferencesRepository;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: field prefix is missing :S

File: data/src/main/java/com/teamagam/gimelgimel/data/user/repository/UserPreferenceRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            editor.putInt(key, (int) value);
+        } else if (value instanceof Boolean) {
+            editor.putBoolean(key, (boolean) value);
+        } else if (value instanceof Float) {
+            editor.putFloat(key, (float) value);
+        } else if (value instanceof Long) {
+            editor.putLong(key, (long) value);
+        } else if (value instanceof String) {
+            editor.putString(key, (String) value);
+        }
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/user/repository/UserPreferenceRepositoryImpl.java]: Maybe we should throw an exception if value doesn't match any if statement

File: domain/src/main/java/com/teamagam/gimelgimel/domain/messages/CreateMessageInteractor.java [hunk 5]
```diff
@@ -28,7 +28,7 @@ protected CreateMessageInteractor(
     protected Observable<T> buildUseCaseObservable() {
         return Observable.create((Subscriber<? super T> subscriber) -> {
             try {
-                String senderId = mUserPreferencesRepository.getSenderId();
+                String senderId = "";//mUserPreferencesRepository.getSenderId();
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/messages/CreateMessageInteractor.java]: What ?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/messages/poller/PolledMessagesProcessor.java [hunk 6]
```diff
@@ -49,7 +49,7 @@ private void processMessage(Message message) {
         }
 
         //Do not broadcast messages from self
-        if (message.getSenderId().equals(mPrefs.getSenderId())) {
+        if (message.getSenderId().equals("")) {
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/messages/poller/PolledMessagesProcessor.java]: What ?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/user/UpdatePreferenceValuesInteractor.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package com.teamagam.gimelgimel.domain.user;
+
+
+public class UpdatePreferenceValuesInteractor {
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/user/UpdatePreferenceValuesInteractor.java]: ?????

**Review by BarAgam — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 3 above)
Comment (teamagam3) [reply app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: Prefix for non-private members?

I didn't know we had this convetion.

File: domain/src/main/java/com/teamagam/gimelgimel/domain/user/UpdatePreferenceValuesInteractor.java (hunk 7 above)
Comment (teamagam3) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/user/UpdatePreferenceValuesInteractor.java]: We have to decide if we use the repository directly or through interactor

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 3 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: Its not private only due to di.
You're right that we don't have a convention for it, but the other injected field in this class is prefixed with m, so we could at least keep it consistent within the class :)

File: domain/src/main/java/com/teamagam/gimelgimel/domain/user/UpdatePreferenceValuesInteractor.java (hunk 7 above)
Comment (BarAgam) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/user/UpdatePreferenceValuesInteractor.java]: yeah well, it's an empty class.

File: data/src/main/java/com/teamagam/gimelgimel/data/user/repository/UserPreferenceRepositoryImpl.java (hunk 4 above)
Comment (BarAgam) [reply data/src/main/java/com/teamagam/gimelgimel/data/user/repository/UserPreferenceRepositoryImpl.java]: what about that?

**Review by BarAgam — APPROVED** [review]

**Pull request merged**